lib/KTable/KTableGridItem.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+
+  <td
+    :class="$computedClass(coreOutlineFocus)"
+    :style="{ textAlign: textAlign, minWidth: minWidth, width: width }"
+    tabindex="0"
+    role="gridcell"
+    @keydown="onKeydown"
+  >
+    <slot :content="content">
+      {{ content }}
+    </slot>
+  </td>
+
+</template>
+
+
+<script>
+
+  import { DATA_TYPE_NUMERIC } from '../composables/useSorting';
+
+  export default {
+    name: 'KTableGridItem',
+    props: {
+      content: {
+        type: [String, Number],
+        required: true,
+      },
+      dataType: {
+        type: String,
+        default: 'string',
+      },
+      rowIndex: {
+        type: Number,
+        required: true,
+      },
+      colIndex: {
+        type: Number,
+        required: true,
+      },
+      minWidth: {
+        type: String,
+        default: 'auto',
+      },
+      width: {
+        type: String,
+        default: 'auto',
+      },
+    },
+    computed: {
+      coreOutlineFocus() {
+        return {
+          ':focus': {
+            ...this.$coreOutline,
+            outlineOffset: '-2px',
+          },
+        };
+      },
+      textAlign() {
+        return this.dataType === DATA_TYPE_NUMERIC ? 'right' : 'left';
+      },
+    },
+    methods: {
+      onKeydown(event) {
```

Comment:
```suggestion
      // Ensures that clickable elements within a table cell, such as buttons and links, can be clicked with ENTER key. 
      onKeydown(event) {
```

`KTable` is a component with the most a11y nuances in our codebase that not many people are familiar with. So I think a few function summary comments like here will be important in long-run, so we know what we're potentially breaking when updating code, and for easier orientation. 

I don't think you need to comment `KTable` file exhaustively, the code is pretty self-explanatory. Perhaps just very tricky/unexpected lines (that you've already commented on), and additionally these functions summaries should do.